.github/workflows/deploy-ghpages-latest-stable.yml

```diff
@@ -0,0 +1,140 @@
+          default: none
+        path-doc:
+          required: false
+          type: string 
+          default: ./doc
+
+
+
+# Sets permissions of the GITHUB_TOKEN to allow deployment to GitHub Pages
+permissions:
+  contents: read
+  pages: write
+  id-token: write
+
+# Allow one concurrent deployment
+concurrency:
+  group: "pages"
+  cancel-in-progress: true
+
+jobs:
+  check-label:
+    runs-on: ubuntu-latest
+    steps:
+      - name: checks for the label 
+        id: valid_label
+        run: |
+          if [[ "${{ inputs.trigger-label }}" != "latest" ]] && [[ "${{ inputs.trigger-label  }}" != "stable" ]];
+            then
+                exit 1 
+          fi
+            
+  
+  build:
+    
+    runs-on: ubuntu-latest
+    needs: check-label
+    steps:
+      
+      - uses: actions/checkout@v3
+      - name: Install poetry
+        run: pipx install poetry
+        if: ${{ inputs.package-manager == 'poetry' }}
+      - uses: actions/setup-python@v4
+        with:
+          python-version: ${{ inputs.python-version }}
+          cache: ${{ inputs.package-manager }}
+          cache-dependency-path: ${{ inputs.dependency-path }} 
+      - name: Install dependencies and compile the documentation with pip
+        run: |
+          pip install .[docs]
+          # required by nbsphinx
+          # https://nbsphinx.readthedocs.io/en/0.2.15/installation.html#pandoc
+          sudo apt install pandoc
+          cd ${{ inputs.path-doc }}
+          make html
+        if : ${{ inputs.package-manager == 'pip' }}
+      - name: Install dependencies and compile the documentation with poetry
+        run: |
+          poetry install --no-interaction ${{ inputs.poetry-extras }}
+          pip install poethepoet
+          poe docs
+        if: ${{ inputs.package-manager == 'poetry' }}
+      - name: Upload artifact
+        uses: actions/upload-artifact@v1
+        with:
+          name: documentation
+          path: ${{ inputs.path-doc }}/build/html
+  
+  # TODO: this job should have a own reusable workflow
+  new-artifact:
+    needs: [build, check-label]
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2 
+    - name: generate the ${{inputs.trigger-label}} documentation 
+      run: |
+        label=${{inputs.trigger-label}}
+        mkdir new_docs
+        cd new_docs
+        mkdir $label
+    - uses: actions/download-artifact@v3
+      with:
+        name: documentation
+        path: new_docs/${{inputs.trigger-label}} 
+    - name: generate the remaning documentation
+      run: | 
+        label=${{inputs.trigger-label}} 
+        cd new_docs
+        if [[ ${label} == "latest" ]]; 
+          then 
+            missing_doc="stable"
+          else
+            missing_doc="latest"
+        fi
+        URL="https://qibogang.github.io/qibo/${missing_doc}/${missing_doc}"
+        mkdir "$missing_doc"
+        # Check if it is possible to connect to the website 
+        if wget "${URL}"; then  
```

Comment:
Sorry @AleCandido , I forgot to remove the `wget` inside the if.
In this way the code works, because the workflow tries to download the url with `wget "${URL}"`, if it fails the `else` branch is applied.
`--spider` is useful only if you don't want to download anything at all, this is not the case.

Reply:
Indeed, I agree this is another possible solution.

I was not proposing to reintroduce `--spider`, just to move the check on the existence of the downloaded file. But this is working as well, and it's less verbose, I like it :)